docs/roadmap.md

```diff
@@ -0,0 +1,169 @@
+architecture as [Index-While-Building][] in XCode 9.
+
+* Goto definition (`textDocument/definition`). Requires
+  [semanticdb-scalac](#semanticdb-scalac).
+  * [x] project -> project.
+  * [x] project -> Scala dependency. Requires [`metalsSetup`](#metalssetup).
+  * [x] project -> Java dependency. Requires [`metalsSetup`](#metalssetup).
+  * [ ] dependency -> dependency.
+* [x] Find references (`textDocument/references`). Works the same as goto
+      definition.
+  * [x] project -> project.
+  * [ ] project -> Scala dependency.
+  * [ ] project -> Java dependency.
+  * [ ] dependency -> dependency.
+* [x] Highlight references to symbol at position
+      (`textDocument/documentHighlight`).
+* [x] Open symbol in file (`textDocument/documentSymbol`)
+* [x] Open symbol in workspace (`workspace/symbol`)
+* [x] Symbol outline in the sidebar as you type (`textDocument/documentSymbol`).
+* [ ] Goto implementation (`textDocument/implementation`)
+* [ ] Goto type definition (`textDocument/typeDefinition`)
+* [x] Show type of symbol at position (`textDocument/hover`). Requires
+      [semanticdb-scalac](#semanticdb-scalac).
+* [ ] Show type of expression at position (`textDocument/hover`).
+
+## Fast completions
+
+Auto completions are a great way to explore a library interface and save
+keystrokes while writing code. Completions are sensitive to latency since
+suggestions update on every keystroke. Correct and fast completions are
+difficult to support in Scala due to several challenges:
+
+1.  the presentation compiler does not lend itself well to interactive editing
+2.  some completions semantics are are difficult to reproduce outside of the
+    compiler
+    * extension methods depend on implicit conversions
+    * whitebox macros can refine result types which influences completion
+      suggestions
+
+It will take a while until Metals can robustly support completions.
+
+* [x] Scala Presentation Compiler. Faithfully reproduces full language semantics
+      but may be slow and difficult to keep in-sync with the build for
+      long-running interactive sessions.
+  * Disabled by default. Can be enabled with configuration option
+    `scalac.completions.enabled=true`.
+  * Requires [`metalsSetup`](#metalssetup).
+  * Shows parameter list as you type (`textDocument/signatureHelp`).
+* [ ] Auto-import global symbol from classpath.
+* [ ] Auto-expand abstract methods when implementing a trait or abstract class.
+
+## Low CPU and memory usage
+
+Heavy resource usage is one of the most frequent complaints about IDEs. Low CPU
+and memory usage becomes even more important when working in larger codebases.
+
+* [ ] Persistent symbol index: currently Metals stores the navigation index
+      in-memory, which can consume multiple GB of RAM for large projects.
+* [ ] Faster Scala and Java outline indexing. Currently, Metals can index the
+      outlines of Scala sources at ~30-40k loc/s and Java sources at ~450k
+      loc/s.
```

Comment:
Is there a concrete goal? Things can always be made faster, but where do we stop?

Reply:
Same as I said before, no funding so I prefer not to make public promises here. If someone is motivated to take on this challenge they will be able to check out these boxes.

It's probably wrong to call this a "roadmap" as it's rather a wishlist

Reply:
It's not a promise though, is it? Elsewhere in the document you're saying "200 ms" as a latency goal for something (don't remember what exactly you were referring to). Why not say here that you want, say ~0.5mloc/s indexing speed?